Add section for official source code (#10678)
mistercrunch/Airflow: README.md

 --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-1.10.12/constraints-3.7.txt"
```

## Official source code

> Looking at the legal side of the software creation process, Apache Airflow is an Apache Software Foundation (ASF) project. This means that Apache Airflow project releases are source code releases and follow the [ASF Release Policy](http://www.apache.org/legal/release-policy.html). There is only one official release repository for Apache Airflow, like all ASF projects: [The Distribution Site of ASF](https://downloads.apache.org). Official releases are cryptographically signed by the release manager and officially voted on by the PMC members during the [Release Approval Process](http://www.apache.org/legal/release-policy.html#release-approval).

Reviewer: Should we do?
```markdown
Apache Airflow is an [Apache Software Foundation (ASF)](http://www.apache.org)
```

Following the ASF rules, the source packages released must be sufficient for a user to build and test the release provided they have access to the appropriate platform and tools.

> Other ways of retrieving source code are "convenience" methods. For example, tagging in GitHub is a convenient way to mark the git project sources that were used to generate the official source packages. We also have binary "convenience" packages how Airflow is distributed: PyPI releases which are a convenient way of installing Apache Airflow using standard python tools or Docker Images published in the [Apache Airflow DockerHub](https://hub.docker.com/repository/docker/apache/airflow). Those are not official releases but they are built using officially released sources.
> 
> > Note: Airflow Summit 2020's "Production Docker Image" talk where context, architecture and customization/extension methods are [explained](https://youtu.be/wDr3Y7q2XoI).

Reviewer 2: ```suggestion
Other ways of retrieving source code are "convenience" methods. For example, tagging in GitHub is a convenient way to mark the git project sources that were used to generate the official source packages. We also have binary "convenience" packages: PyPI releases which are a convenient way of installing Apache Airflow using standard python tools or Docker Images published in the [Apache Airflow DockerHub](https://hub.docker.com/repository/docker/apache/airflow). Those are not official releases but they are built using officially released sources.

> Note: Airflow Summit 2020's "Production Docker Image" talk where context, architecture and customization/extension methods are [explained](https://youtu.be/wDr3Y7q2XoI).
```

Reviewer 2: LGTM otherwise. I am happy with the wording; making the README concise and small (if other's think so too ) can be a separate issue/PR.

Reviewer 3: Fully Agree. The smaller and more concise README we can have, the better. It should have all the relevant links to other docs (maybe an automated checs if all the .md /.rst files are reachable from the README) and should have  only most important info. It can be indeed separate issue though (but we this one we are already decreasing the readme a bit)

 --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-1.10.12/constraints-3.7.txt"
```

> ## Official source code

Reviewer 3: ```suggestion
## Official Airflow releases
```

> There is only one "official" source code repository for Apache Airflow, like all ASF projects: https://downloads.apache.org. Official releases are cryptographically signed by the release manager and placed in the Apache svn next to the file, its checksum and archived by ASF in case of any legal disputes.

Reviewer 4: Typo around `...its checksum and archived by ASF...`?

Author: Thanks - fixed :)

> There is only one "official" source code repository for Apache Airflow, like all ASF projects: https://downloads.apache.org. Official releases are cryptographically signed by the release manager and placed in the Apache svn next to the file, with the checksum archived by ASF in case of any legal disputes.

Reviewer 3: ```suggestion
Looking at the legal side of the software creation process, Apache Airflow is an Apache Sofware Foundation (ASF) project. This means that Apache Airflow project releases are source code releases and follow the [ASF Release Policy](http://www.apache.org/legal/release-policy.html). There is only one official release repository for Apache Airflow, like all ASF projects: [The Distribution Site of ASF](https://downloads.apache.org). Official releases are cryptographically signed by the release manager and officially voted on by the PMC members during the [Release Approval Process](http://www.apache.org/legal/release-policy.html#release-approval).```

> Other ways of retrieving source code are just "convenience" methods. For example, tagging in GitHub is a convenient way to mark the git project sources that were used to generate the "official" source package. What is in the Github tag is not a concern, what matters is what goes into the official package.

Reviewer 3: ```suggestion
Following the ASF rules, the source packages released must be sufficient for a user to build and test the release provided they have access to the appropriate platform and tools. 

Other ways of retrieving source code are "convenience" methods. For example, tagging in GitHub is a convenient way to mark the git project sources that were used to generate the official source packages. We also have binary "convenience" packages how Airflow is distributed: PyPI releases which are a convenient way of installing Apache Airflow using standard python tools or Docker Images published in the [Apache Airflow DockerHub](https://hub.docker.com/repository/docker/apache/airflow). Those are not official releases but they are built using officially released sources.
```

> ## Official source code
> 
> There is only one "official" source code repository for Apache Airflow, like all ASF projects: https://downloads.apache.org. Official releases are cryptographically signed by the release manager and placed in the Apache svn next to the file, with the checksum archived by ASF in case of any legal disputes.
> 
> Other ways of retrieving source code are just "convenience" methods. For example, tagging in GitHub is a convenience way to mark the git project sources that were used to generate the "official" source package. What is in the Github tag is not a concern, what matters is what goes into the official package.

Reviewer 2: ```suggestion
## Official source code

There is only one "official" source code repository for Apache Airflow, like all ASF projects: https://downloads.apache.org. Official releases are cryptographically signed by the release manager and placed in the Apache svn next to the file, with the checksum archived by ASF in case of any legal disputes.

Other ways of retrieving source code are just "convenience" methods. For example, tagging in GitHub is a convenient way to mark the git project sources that were used to generate the "official" source package. What is in the Github tag is not a concern, what matters is what goes into the official package.
```

> Other ways of retrieving source code are just "convenience" methods. For example, tags in GitHub is only a convenience way to mark that those are the git project sources that were used to generate the "official" source package. What is in the tag is not a concern when it comes to releasing the software, but what matters most is what goes into the official package.

Reviewer 4: maybe typos for `tags`/`tag` and `is`/`are`?

Author: Thanks - fixed :)

Following the ASF rules, the source packages released must be sufficient for a user to build and test the release provided they have access to the appropriate platform and tools. 

Other ways of retrieving source code are "convenience" methods. For example, tagging in GitHub is a convenient way to mark the git project sources that were used to generate the official source packages. We also have binary "convenience" packages how Airflow is distributed: PyPI releases which are a convenient way of installing Apache Airflow using standard python tools or Docker Images published in the [Apache Airflow DockerHub](https://hub.docker.com/repository/docker/apache/airflow). Those are not official releases but they are built using officially released sources.

> ## Official Docker images

Reviewer 3: I think we ccan also skip this whole section here. It was useful when we were working on the official images, but if we replace with the above mention, it should be enough.
```suggestion
```

Author: ok deleted that section

> In order to use Airflow in Docker Compose or Kubernetes, you might need to use or build production images of Apache Airflow. The production image is a multi-segment image. The first segment "airflow-build-image" contains all the build essentials and related dependencies that allow to install airflow locally. By default the image is build from a released version of Airflow from Github, but by providing some extra arguments you can also build it from local sources.

Reviewer 3: ```suggestion
```

>  This is particularly useful in CI environment where we are using the image to run Kubernetes tests(Helm Chart integration). We will also use released images in the Helm Chart(backward compatibility). You can see DockerHub images at https://hub.docker.com/repository/docker/apache/airflow. In DockerHub there is just a convenience binary and that image can (and often should) be built from the officially released sources. More Details [TODO](#).

Reviewer 3: ```suggestion
```
